### ISSUE TITLE
Notepad++ does not recognize php functions that include an odd number of curly brackets

### ISSUE THREAD

**Issue by miile7** [notepad-plus-plus/notepad-plus-plus#4133]
### Notepad++ does not recognize php functions that include an odd number of curly brackets
Notepad++ does not recognize php functions (in the function panel) with any modifier e.g. `public`, `protected`, `static`, `final`, ... if there is an odd number of curly brackets in a String.

### Code example:
In any class the code
```
public function a(){
    return "{";
}
```
does not work while
```
public function a(){
    return "{}";
}
```
works.
Also a normal function which is not located in a class and does not have any modifiers works with an odd number of curly brackets.

**Comment by operasfantom** [comment]
See #3910

**Comment by MAPJe71** [comment]
@miile7 Please try my updated PHP parser posted [here](https://notepad-plus-plus.org/community/topic/15124/php-function-list-and-abstract-functions/11).

**Comment by ShoresOfNowhere** [comment]
No, the linked updated php parser doesn't solve this problem...

Simply put, the parser should totally ignore curly braces in quoted or double quoted strings...

Any hope of getting this resolved? :)

**Comment by MAPJe71** [comment]
@ShoresOfNowhere 
> Simply put, the parser should totally ignore curly braces in quoted or double quoted strings...

The parser does ignore everything between single and double quotes strings.

There might be more than one `functionList.xml` on you system. 
Update the parser in the "active" one and check again with aforementioned code example.

**Comment by ShoresOfNowhere** [comment]
You're right! I didn't notice the functionList.xml file is also present in the user profile folder.

Now all works properly!

Thanks!
